src/Libraries/PythonNodeModels/PythonNode.cs

```diff
@@ -49,7 +51,8 @@ public PythonEngineVersion Engine
                 if (engine != value)
                 {
                     engine = value;
-                    RaisePropertyChanged("Engine");
+                    RaisePropertyChanged(nameof(Engine));
+                    OnNodeModified();
```

Comment:
@mjkkirschner @aparajit-pratap this will mark the node as modified when the engine property is modified. Also, the workspace will have unsaved changes.